### DISCUSSION TITLE
how to use brew installed llvm to compile brew package?

### DISCUSSION THREAD

**Discussion by htzh** [Homebrew/discussions#2151] [Everyday usage] [answered]
I need the brew version of llvm for OpenMP support. After installing brew llvm, even with priority in $PATH, brew still chose the system compiler I believe:
```
CMakeOutput.log
```
```
The system is: Darwin - 20.6.0 - arm64
Compiling the CXX compiler identification source file "CMakeCXXCompilerId.cpp" succeeded.
Compiler: /opt/homebrew/Library/Homebrew/shims/mac/super/clang++ 
```

How can I force brew to use the brew installed llvm compilers for this particular package?

**Comment by htzh** [comment]
If I set ```HOMEBREW_CC=llvm_clang++``` I get:

```
RuntimeError: Invalid value for HOMEBREW_CC: llvm_clang++
```

**Reply by htzh** [reply]
Changing to ```HOMEBREW_CXX=llvm_clang++``` eliminated the exception but the compiler chosen is still the system compiler.

**Comment by htzh** [comment, marked as answer]
Third try is charm: ```HOMEBREW_CC=llvm_clang``` appears to work for ```CXX``` as well. This seemed to have resulted in the correct compiler being picked. I still have other problems to resolve but at least this question seemed to have an answer. The fact that ```HOMEBREW_CXX=llvm_clang++``` does not work as intended looks like a bug to me.

**Comment by carlocab** [comment]
What formula do you need OpenMP support for?

**Reply by htzh** [reply]
I need it for [Souffle](https://souffle-lang.github.io/build). Without OpenMP it works but it would work a lot better with OpenMP.

**Reply by htzh** [reply]
The installation was done using

```
brew install --HEAD souffle-lang/souffle/souffle
```
